src/main/java/ch/njol/skript/expressions/ExprIndicesOfX.java

```diff
@@ -0,0 +1,128 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.lang.Variable;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Name("Indices of X in List")
+@Description("Returns all the indices of a list where the value at that index is the provided value.")
+@Examples({
+	"set {_list::*} to 1, 2, 3, 1, 2, 3",
+	"set {_indices::*} to the indices of the value 1 in {_list::*}",
+	"# {_indices::*} is now \"1\", \"4\"",
+	"",
+	"set {_indices::*} to the indices of the value 2 in {_list::*}",
+	"# {_indices::*} is now \"2\", \"5\"",
+	"",
+	"set {_otherlist::burb} to 100",
+	"set {_otherlist::burp} to 100",
+	"set {_indices::*} to the first index of the value 100 in {_otherlist::*}",
+	"# {_indices::*} is now \"burb\"",
+	"set {_indices::*} to the last index of the value 100 in {_otherlist::*}",
+	"# {_indices::*} is now \"burp\""
+})
+@Since("INSERT VERSION")
+public class ExprIndicesOfX extends SimpleExpression<String> {
+
+	static {
+		Skript.registerExpression(ExprIndicesOfX.class, String.class, ExpressionType.COMBINED,
+			"[the] [1:first|2:last] (indices|index[es]) of [[the] value] %object% in %objects%"
+		);
+	}
+
+	private IndexType type;
+	private Expression<?> value;
+	private Variable<?> list;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!(exprs[1] instanceof Variable<?> var) || !var.isList()) {
+			Skript.error("'" + exprs[1].toString(null, false) +
+				"' can only ever have one value at most, thus the 'indices of x in ...' expression is useless.");
```

Comment:
Is this true? Can't a user pass in an expression here which returns multiple values which isn't a variable?

Reply:
it checks if it's a variable & list beforehand and I think it should only work with variables.

Reply:
you are indeed right